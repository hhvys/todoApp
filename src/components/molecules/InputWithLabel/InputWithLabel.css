.div__input__label {
    height: 45px;
    background-color: #7A9676;
    color: #FFFFFF;
    border-radius: 5px;
}

.div__input__label label {
    margin: 10px;
    margin-bottom: 14px;
    cursor: pointer;
    font-size: 33px;
    flex: 1;
}

