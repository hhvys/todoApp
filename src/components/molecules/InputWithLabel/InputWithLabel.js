import React from 'react';
import styles from './InputWithLabel.mod.scss';
import Input from "../../atoms/Input/Input";

const handleKeyUp = (e, onSubmit) => {
	if (e.nativeEvent.keyCode === 13) {
		const inputValue = e.nativeEvent.target.value.trim();
		if (inputValue)
			onSubmit(inputValue);
		e.nativeEvent.target.value = '';
	}
};

const InputWithLabel = ({
													className,
													label,
													placeholder,
													onInputChange,
													onSubmit,
													...props
												}) => (
	<div {...props}
			 className={`${className ? className : ''} ${styles.inputLabel} full-width d-flex align-items-center justify-content-center`}>
		<label
			className={`${styles.addSymbol} full-height d-flex align-items-center justify-content-center`}>{label}</label>
		<Input onInputChange={() => {
		}}
					 onKeyUp={(e) => handleKeyUp(e, onSubmit)}
					 placeholder={placeholder}
		/>
	</div>
);

export default InputWithLabel;
