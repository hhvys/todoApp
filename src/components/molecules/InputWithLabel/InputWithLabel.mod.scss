.inputLabel {
    height: 45px;
    background-color: #7A9676;
    color: #FFFFFF;
    border-radius: 5px;
}

.addSymbol {
    margin: 10px 10px 14px;
    cursor: pointer;
    font-size: 33px;
    width: 25px;
}

