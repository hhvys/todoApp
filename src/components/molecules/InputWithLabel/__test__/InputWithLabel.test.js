import React from 'react';
import InputWithLabel from '../InputWithLabel';

describe('InputWithLabel', () => {

	let props;
	let mountedInputWithLabel;

	const inputWithLabel = () => {
		if(!mountedInputWithLabel) {
			mountedInputWithLabel = shallow(
				<InputWithLabel {...props}/>
			);
		}
		return mountedInputWithLabel;
	};

	beforeEach(() => {
		props = {
			className: undefined,
			label: undefined,
			placeholder: undefined,
			onInputChange: undefined,
			onSubmit: undefined,
		};
		mountedInputWithLabel = undefined;
	});

	it('should pass className prop to wrapper div', () => {
		props.className = "randomClass";
		const divs = inputWithLabel().find('div');
		const wrapperDiv = divs.first();
		expect(toJSON(wrapperDiv)).toMatchSnapshot();
	});

	it('should not render label if label is undefined or null', () => {
		const wrapperDiv = inputWithLabel().find('div').first();
		expect(toJSON(wrapperDiv)).toMatchSnapshot();
	});

	it('should render label if label prop is defined', () => {
		props.label="anyLabel";
		const wrapperDiv = inputWithLabel().find('div').first();
		expect(toJSON(wrapperDiv)).toMatchSnapshot();
	});

	it('should pass placeholder prop to Input', () => {
		props.placeholder = "placeholder";
		const Input = inputWithLabel().find('Input').first();
		expect(toJSON(Input)).toMatchSnapshot();
	});

	it('should pass onInputChange prop to Input if onInputChange is defined', () => {
		props.onInputChange = jest.fn();
		const Input = inputWithLabel().find('Input').first();
		expect(Input.props().onInputChange).toBe(props.onInputChange);
	});

	it('should pass () => {} as onInputChange to Input if onInputChange is not defined', () => {
		const Input = inputWithLabel().find('Input').first();
		expect(Input.props().onInputChange).toString('() => {}');
	});
});
