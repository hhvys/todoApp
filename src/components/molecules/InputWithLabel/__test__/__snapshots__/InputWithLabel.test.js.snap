// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel should not render label if label is undefined or null 1`] = `
<div
  className=" inputLabel full-width d-flex align-items-center justify-content-center"
>
  <Input
    onInputChange={[Function]}
    onKeyUp={[Function]}
  />
</div>
`;

exports[`InputWithLabel should pass className prop to wrapper div 1`] = `
<div
  className="randomClass inputLabel full-width d-flex align-items-center justify-content-center"
>
  <Input
    onInputChange={[Function]}
    onKeyUp={[Function]}
  />
</div>
`;

exports[`InputWithLabel should pass placeholder prop to Input 1`] = `
<Input
  onInputChange={[Function]}
  onKeyUp={[Function]}
  placeholder="placeholder"
/>
`;

exports[`InputWithLabel should render label if label prop is defined 1`] = `
<div
  className=" inputLabel full-width d-flex align-items-center justify-content-center"
>
  <label
    className="addIcon full-height d-flex align-items-center justify-content-center"
  >
    anyLabel
  </label>
  <Input
    onInputChange={[Function]}
    onKeyUp={[Function]}
  />
</div>
`;
